### PR TITLE
Add phantomjs

### LINKED ISSUE #119: Make travis run the UI tests
This is an issue for people who would like to learn basic quality assurance using tests for web applications.
**I'm submitting a ...**
- [X] bug report

**Current behavior:**

Currently, travis does not run the UI tests.
These are tests that click on the webpage of the application and make sure, certain uer/stories and work flows stay while we develop.

**Expected behavior:**

I would liketravis to run those tests.

- [ ] add the tests to travis
- [ ] see that they fail in your pull-request because firefox can not start because travis has no GUI
- [ ] install phantomJS, a browser for such purposes without GUI and create a central place to change the "driver" (see source code)

**Steps to reproduce:**

https://travis-ci.org/fossasia/badgeyay/builds/277071140#L563
Related: a discussion in #116

### PR DESCRIPTION
**Pull request by S2606**
Fixes #119 

#### Checklist

- [X] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [X] My branch is up-to-date with the Upstream `development` branch.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by S2606** [comment]
@gabru-md Please review

**Comment by jajodiaraghav** [comment]
On visiting the Phantom.js website, it says that we have to manually install phantom.js to use it. 
How can we integrate this to Travis?

**Comment by jajodiaraghav** [comment]
@gabru-md Please see if this is the right way to integrate phantomjs

**Comment by S2606** [comment]
@jajodiaraghav Travis page says that phantomjs is alerady embedded in their root folder

File: .travis.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ before_install:
 - sudo pip install -r test-requirements.txt
 
 script:
-- python app/tests/test.py
+- phantomjs app/tests/test.py
```
Comment (niccokunzmann) [inline .travis.yml]: Maybe this change is not required.

Comment (gabru-md) [reply .travis.yml]: Yes i do believe the same. 
@S2606 please look for it.

Comment (niccokunzmann) [reply .travis.yml]: phantomjs is a browser which can be used by many programming languages to test the code.
I do not think it understands the python file argument as running the file with python, more like visiting it as web page.

File: app/tests/test.py [hunk 2]
```diff
@@ -7,7 +7,7 @@ class BadgeyayTest(unittest.TestCase):
 
 	@classmethod
 	def setUpClass(cls):
-		cls.driver = webdriver.Firefox()
+		cls.driver = webdriver.PhantomJS()
```
Comment (niccokunzmann) [inline app/tests/test.py]: I think, this will do it.

**Comment by S2606** [comment]
@niccokunzmann @gabru-md Ok i will make the changes.
@jajodiaraghav I will make a new commit since force pushing on the same commit crashes travis CI

**Comment by S2606** [comment]
@niccokunzmann @jajodiaraghav @gabru-md Please review

**Comment by niccokunzmann** [comment]
Whohoo, test runs...

**Comment by niccokunzmann** [comment]
Thanks! I will merge. The problem remains that I do not see a test output.
https://travis-ci.org/fossasia/badgeyay/builds/278113797#L567

**Comment by S2606** [comment]
@niccokunzmann maybe this lnk can help
https://ilovesymposia.com/2014/10/15/continuous-integration-in-python-5-report-test-coverage-using-coveralls/

**Pull request merged**